src/main/java/duke/logic/Storage.java

```diff
@@ -0,0 +1,106 @@
+package duke.logic;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+
+import duke.DukeException;
+import duke.logic.task.Deadline;
+import duke.logic.task.Event;
+import duke.logic.task.Task;
+import duke.logic.task.Todo;
+
+/**
+ * Handle loading and saving of data.
+ */
+public class Storage {
+
+    protected String filePath;
+
+    /**
+     * Constructor for Storage object.
+     *
+     * @param filePath String containing the file path of data text file.
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Creates Tasks based on data text file at filePath and adds into TaskList.
+     *
+     * @param taskList TaskList which tasks should be loaded into.
+     * @throws DukeException If error reading from file.
+     */
+    public void load(TaskList taskList) throws DukeException {
+        try {
+            FileReader fileReader = new FileReader(this.filePath);
+            BufferedReader bufferedReader = new BufferedReader(fileReader);
+            String line = bufferedReader.readLine();
+            while (line != null) {
```

Comment:
Hmm maybe the logical units within this while block could be separated better? e.g. lines 57 and 58 could be swapped, and a space could be added between lines 50 and 51.

Descriptions of each logical unit could also be added for clarity!